vertx-consul-client/src/main/java/io/vertx/ext/consul/NodeQueryOptions.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2016 The original author or authors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.ext.consul;
+
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Options used to requesting list of nodes
+ *
+ * @author <a href="mailto:ruslan.sennov@gmail.com">Ruslan Sennov</a>
+ */
+@DataObject
+public class NodeQueryOptions {
+
+  private static final String NEAR_KEY = "Near";
+  private static final String BLOCK_KEY = "Block";
+
+  private String near;
+  private BlockingQueryOptions options;
+
+  /**
+   * Default constructor
+   */
+  public NodeQueryOptions() {}
+
+  /**
+   * Constructor from JSON
+   *
+   * @param json the JSON
+   */
+  public NodeQueryOptions(JsonObject json) {
+    near = json.getString(NEAR_KEY);
+    options = json.containsKey(BLOCK_KEY) ? new BlockingQueryOptions(json.getJsonObject(BLOCK_KEY)) : null;
+  }
+
+  /**
+   * Convert to JSON
+   *
+   * @return the JSON
+   */
+  public JsonObject toJson() {
```

Comment:
oupsy, missed it sorry. Converters are classes generated at compile-time managing the serialization/de-serialization of data objects. It's not mandatory, but it's a nice feature. Look at https://github.com/vert-x3/vertx-configuration/blob/initial-work/vertx-configuration/src/main/java/io/vertx/ext/configuration/ConfigurationRetrieverOptions.java for an example.

Again it's not mandatory, but it standardize the json mapping.